engine/src/widget-events.cpp

```diff
@@ -496,26 +508,121 @@ void MCWidgetEventManager::event_paint(MCWidget *p_widget, MCGContextRef p_gcont
     MCWidgetOnPaint(p_widget -> getwidget(), p_gcontext);
 }
 
-void MCWidgetEventManager::event_touch(MCWidget* p_widget, uint32_t p_id, MCEventTouchPhase p_phase, int2 p_x, int2 p_y)
+Bool MCWidgetEventManager::event_touch(MCWidget* p_widget, uint32_t p_id, MCEventTouchPhase p_phase, int2 p_x, int2 p_y)
 {
+    if (!widgetIsInRunMode(p_widget))
+    {
+        return False;
+    }
+    
+    /* If a touch is just starting, and this is the first touch then we need
+     * to compute the touched widget. */
+    MCWidgetRef t_touched_widget = m_touched_widget;
+    if (p_phase == kMCEventTouchPhaseBegan)
+    {
+        if (t_touched_widget == nullptr)
+        {
+            t_touched_widget = hitTest(p_widget->getwidget(), p_x, p_y);
```

Comment:
@runrevmark hmm... so this means that multi-touch if the user is touching other objects only the first touch target will get events? I guess it makes sense seeing as the alternative is multiple objects thinking they are getting single touched

Reply:
Yes - that's correct. LCS isn't quite right at the moment - the control accepting the first touch is actually what should receive subsequent ones otherwise per-control multi-touch gestures won't work unless all touches hit-test within the widget (which I don't think is what is expected by a user).